### PR TITLE
added test for application creation using Store APIs with SSO.

### PR DESCRIPTION
**Pull request by ruks**
added test for application creation using Store APIs with SSO.
Public Jira: https://wso2.org/jira/browse/APIMANAGER-4005

### CODE REVIEW

File: modules/integration/tests-platform/src/test/java/org/wso2/automation/platform/tests/apim/is/SingleSignOnTestCase.java [hunk 1]
```diff
@@ -604,14 +609,53 @@ private void createApplication() throws Exception {
...
+
+        urlParameters.clear();
+        urlParameters.add(new BasicNameValuePair("action", "removeApplication"));
+        urlParameters.add(new BasicNameValuePair("application", testApplicationName));
+        response = sendPOSTMessageWithCSRF(httpsStoreUrl + "/site/blocks/application/" +
+                "application-remove/ajax/application-remove.jag", urlParameters, csrf);
+        assertEquals(Response.Status.OK.getStatusCode(), response.getStatusLine().getStatusCode(),
+                "Response mismatch not 200");
+        boolean errorOccur = getResponseBody(response).contains("\"error\" : true");
+        assertTrue(!errorOccur, "Error on Application deletion");
```
Comment (harsha89) [inline modules/integration/tests-platform/src/test/java/org/wso2/automation/platform/tests/apim/is/SingleSignOnTestCase.java]: Could have use assertFalse

File: modules/integration/tests-platform/src/test/java/org/wso2/automation/platform/tests/apim/is/SingleSignOnTestCase.java [hunk 2]
```diff
@@ -604,14 +609,53 @@ private void createApplication() throws Exception {
...
-        response = sendPOSTMessage(httpsStoreUrl + "/site/blocks/application/" +
-                                   "application-add/ajax/application-add.jag",
-                                   urlParameters);
+        urlParameters.add(new BasicNameValuePair("application", testApplicationName));
+        response = sendPOSTMessageWithCSRF(httpsStoreUrl + "/site/blocks/application/" +
+                "application-add/ajax/application-add.jag", urlParameters, csrf);
         assertEquals(Response.Status.OK.getStatusCode(), response.getStatusLine().getStatusCode(),
-                     "Response mismatch not 200");
+                "Response mismatch not 200");
+        boolean errorOccur = getResponseBody(response).contains("\"error\" : true");
```
Comment (harsha89) [inline modules/integration/tests-platform/src/test/java/org/wso2/automation/platform/tests/apim/is/SingleSignOnTestCase.java]: how about only using boolean error or boolean isError

File: modules/integration/tests-platform/src/test/java/org/wso2/automation/platform/tests/apim/is/SingleSignOnTestCase.java (hunk 1 above)
Comment (ruks) [inline modules/integration/tests-platform/src/test/java/org/wso2/automation/platform/tests/apim/is/SingleSignOnTestCase.java]: yes. it is more meaningful.
Thanks.

File: modules/integration/tests-platform/src/test/java/org/wso2/automation/platform/tests/apim/is/SingleSignOnTestCase.java (hunk 2 above)
Comment (ruks) [inline modules/integration/tests-platform/src/test/java/org/wso2/automation/platform/tests/apim/is/SingleSignOnTestCase.java]: yes. is error is more simple and meaningful.
Thanks.

File: modules/integration/tests-platform/src/test/java/org/wso2/automation/platform/tests/apim/is/SingleSignOnTestCase.java [hunk 3]
```diff
@@ -151,6 +153,7 @@ public void init() throws APIManagerIntegrationTestException {
     @AfterClass(alwaysRun = true)
     public void destroy() throws Exception {
         super.cleanup();
+        deleteApplication();
```
Comment (tharindu1st) [inline modules/integration/tests-platform/src/test/java/org/wso2/automation/platform/tests/apim/is/SingleSignOnTestCase.java]: move this line before above line, After super.cleanup method couldn't do any of the server requests.

Comment (ruks) [inline modules/integration/tests-platform/src/test/java/org/wso2/automation/platform/tests/apim/is/SingleSignOnTestCase.java]: okay. fixed.
Thanks.

File: modules/integration/tests-platform/src/test/java/org/wso2/automation/platform/tests/apim/is/SingleSignOnTestCase.java [hunk 4]
```diff
@@ -91,6 +92,7 @@
     private String apiName = "SingleSignOnAPI";
     private String apiVersion = "1.0.0";
     private String callbackUrl = "www.youtube.com";
+    private String testApplicationName = "SSOTestApplication";
```
Comment (Jasintha) [inline modules/integration/tests-platform/src/test/java/org/wso2/automation/platform/tests/apim/is/SingleSignOnTestCase.java]: Java standard constant naming convention should be followed here

**Pull request merged**